### PR TITLE
feat: Add SectionsToHide to SectionVisibilityFilter

### PR DESCRIPTION
**Pull request by bartlomiejmarszal**
In order to hide the explicit sections but keep them displed as default, I'm introducing a new option in SectionVisibilityFilter config. 

TODO:
Add new version of tao-core to composer requirement.

### CODE REVIEW

File: models/classes/menu/SectionVisibilityFilter.php [hunk 1]
```diff
@@ -31,16 +31,20 @@ class SectionVisibilityFilter extends ConfigurableService implements SectionVisi
...
      * @throws LogicException
      */
     public function isVisible(string $section): bool
     {
         $sections = $this->getOption(self::OPTION_FEATURE_FLAG_SECTIONS);
+        $sectionToHide = $this->getOption(self::OPTION_FEATURE_FLAG_SECTIONS_TO_HIDE);
 
-        if (empty($sections[$section])) {
-            return true;
+        foreach ($sectionToHide[$section] as $featureFlag) {
```
Comment (andreluizmachado) [inline models/classes/menu/SectionVisibilityFilter.php:44]:
```suggestion
        foreach ($sectionToHide[$section] ?? [] as $featureFlag) {
```

File: models/classes/menu/SectionVisibilityFilter.php [hunk 2]
```diff
@@ -31,16 +31,20 @@ class SectionVisibilityFilter extends ConfigurableService implements SectionVisi
...
+            if ($this->getFeatureFlagChecker()->isEnabled($featureFlag)) {
```
Comment (andreluizmachado) [inline models/classes/menu/SectionVisibilityFilter.php:45]:
```suggestion
           return !$this->getFeatureFlagChecker()->isEnabled($featureFlag);
```

Comment (gabrielfs7) [reply models/classes/menu/SectionVisibilityFilter.php:45]: We cannot do that cause we need to iterate all the featureFlags

File: models/classes/menu/SectionVisibilityFilter.php [hunk 3]
```diff
@@ -31,16 +31,20 @@ class SectionVisibilityFilter extends ConfigurableService implements SectionVisi
 {
     public const SERVICE_ID = 'tao/SectionVisibilityFilter';
     public const OPTION_FEATURE_FLAG_SECTIONS = 'featureFlagSections';
+    public const OPTION_FEATURE_FLAG_SECTIONS_TO_HIDE = 'featureFlagSectionsToHide';
```
Comment (gabrielfs7) [inline models/classes/menu/SectionVisibilityFilter.php:34]: Please also document this new option here https://github.com/oat-sa/tao-core/tree/master/models/classes/featureFlag#sectionvisibilityfilterinterface

File: models/classes/featureFlag/FeatureFlagCheckerInterface.php [hunk 4]
```diff
@@ -28,6 +28,7 @@ interface FeatureFlagCheckerInterface
     public const FEATURE_FLAG_TABULAR_IMPORT = 'FEATURE_FLAG_TABULAR_IMPORT_ENABLED';
     public const FEATURE_FLAG_LISTS_DEPENDENCY_ENABLED = 'FEATURE_FLAG_LISTS_DEPENDENCY_ENABLED';
     public const FEATURE_FLAG_STATISTIC_METADATA_IMPORT = 'FEATURE_FLAG_STATISTIC_METADATA_IMPORT';
+    public const FEATURE_FLAG_PASSWORD_CHANGE_DISABLED = 'FEATURE_FLAG_PASSWORD_CHANGE_DISABLED';
```
Comment (gabrielfs7) [inline models/classes/featureFlag/FeatureFlagCheckerInterface.php:31]: Please also document it here https://github.com/oat-sa/tao-core/blob/master/README.md#environment-variables

File: models/classes/menu/SectionVisibilityFilter.php [hunk 5]
```diff
@@ -31,16 +31,20 @@ class SectionVisibilityFilter extends ConfigurableService implements SectionVisi
...
      * @throws LogicException
      */
     public function isVisible(string $section): bool
     {
         $sections = $this->getOption(self::OPTION_FEATURE_FLAG_SECTIONS);
+        $sectionToHide = $this->getOption(self::OPTION_FEATURE_FLAG_SECTIONS_TO_HIDE);
 
-        if (empty($sections[$section])) {
-            return true;
+        foreach ($sectionToHide[$section] ?? [] as $featureFlag) {
```
Comment (gabrielfs7) [inline models/classes/menu/SectionVisibilityFilter.php:44]: Can you please apply this on the line `50`? 

```
$sections[$section] ?? []
```

**Review by gabrielfs7 — CHANGES_REQUESTED** [review]
Thanks @bartlomiejmarszal !

Almost all good, we are missing to add documentation and some extra validation.

File: models/classes/menu/SectionVisibilityFilter.php (hunk 5 above)
Comment (bartlomiejmarszal) [reply models/classes/menu/SectionVisibilityFilter.php:44]: I will define default value on `getOption` method instead.

Comment (gabrielfs7) [reply models/classes/menu/SectionVisibilityFilter.php]: This will not solve the issue, cause you are evaluation section key existence

**Review by gabrielfs7 — APPROVED** [review]
Thanks! Good job @bartlomiejmarszal 

- [x] New code is covered by tests (if applicable)
- [x] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful

**Pull request merged**